Standards/scs-0103-v1-standard-flavors.md

```diff
@@ -0,0 +1,134 @@
+---
+title: SCS Standard Flavors
+type: Standard
+status: Draft
+track: IaaS
+---
+
+## Introduction
+
+## Motivation
+
+In OpenStack environments there is a need to define different flavors for instances.
+The flavors are pre-defined by the operator, the customer can not change these.
+OpenStack providers thus typically offer a large selection of flavors.
+
+While flavors can be discovered (`openstack flavor list`), it is helpful for users (DevOps teams),
+to have a guaranteed set of flavors available on all SCS clouds, so these need not be discovered.
+
+## Standard SCS flavors
+
+These are flavors that must exist on standard SCS clouds (x86-64).
+
+### Mandatory
+
+| Recommended name | vCPUs  | vCPU type     | RAM [GiB]  | Root disk [GB]  | Disk type  |
+| ---------------- | ------ | ------------- | ---------- | --------------- | ---------- |
+| SCS-1V-4         |      1 | shared core   |          4 |                 |            |
+| SCS-2V-8         |      2 | shared core   |          8 |                 |            |
+| SCS-4V-16        |      4 | shared core   |         16 |                 |            |
+| SCS-8V-32        |      8 | shared core   |         32 |                 |            |
+| SCS-1V-2         |      1 | shared core   |          2 |                 |            |
+| SCS-2V-4         |      2 | shared core   |          4 |                 |            |
+| SCS-4V-8         |      4 | shared core   |          8 |                 |            |
+| SCS-8V-16        |      8 | shared core   |         16 |                 |            |
+| SCS-16V-32       |     16 | shared core   |         32 |                 |            |
+| SCS-1V-8         |      1 | shared core   |          8 |                 |            |
+| SCS-2V-16        |      2 | shared core   |         16 |                 |            |
+| SCS-4V-32        |      4 | shared core   |         32 |                 |            |
+| SCS-1L-1         |      1 | crowded core  |          1 |                 |            |
```

Comment:
**TODO**: Resolve the following issues:

- We say what resources the flavors should offer. We don't say what resources should NOT be offered. For instance, none of the flavors explicitly offers a GPU, but we don't disallow a GPU either. However, for a customer, this may make a big difference, not least with regards to pricing. (**Note that this issue exists with the original standard as well.**)
- We say that the mandatory and recommended flavors must come with certain `extra_specs`. We have to say something about the converse: which (other) flavors may carry (any, some subset, or all of) these `extra_specs`. For the time being, I suppose we require that the presence of any one of these properties imply that the flavor is one of those listed here (and consequently, the other properties must be present as well). In the medium run, we have to define in general which of these properties are compulsory for any flavor.

Reply:
> We say that the mandatory and recommended flavors must come with certain extra_specs. We have to say something about the converse: which (other) flavors may carry (any, some subset, or all of) these extra_specs.

I disagree, we don't have to say anything about the converse, because there is an infinite number of combinations of flavors and extra_specs. We only define and standardise what we define, everything else is up to the CSP.

Reply:
@artificial-intelligence The problem is that we want discoverability. People want to find their flavor by looking at extra_specs. So when some flavor carries these extra_specs, that has to mean something!